src/style_layer.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+var util = require('./util');
+var format_url = require('./format_url');
+var request = require('./request');
+
+var StyleLayer = L.TileLayer.extend({
+    includes: [require('./load_tilejson')],
+
+    options: {
+        sanitizer: require('sanitize-caja')
+    },
+
+    initialize: function(_, options) {
+        L.TileLayer.prototype.initialize.call(this, undefined, options);
+
+        this.options.tiles = this._formatTileURL(_);
+
+        this._tilejson = {};
+
+        this._getAttribution(_);
+    },
+
+    _setTileJSON: function(json) {
+        util.strict(json, 'object');
+
+        L.extend(this.options, {
+            attribution: this.options.sanitizer(json.attribution),
+            tms: false,
```

Comment:
There are many [`L.TileLayer` options](http://leafletjs.com/reference.html#tilelayer) that will cause this class to malfunction if they are misused. Why call out `tms` specifically?